frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+      this.setState({ readOnly });
+    };
+
+    onFormDisable = disable => {
+      this.setState({
+        disableForm: disable,
+      });
+    };
+
+    isValidForm = () => {
+      // Check that a valid PVC has been choosen
+      if (!this.state.formData.claimName) {
+        this.setState({ error: 'Storage must be selected', inProgress: false });
+        return false;
+      }
+      return true;
+    };
+
+    // Add Attach storage //
+    save = (event: React.FormEvent<EventTarget>) => {
+      event.preventDefault();
+      if (!this.isValidForm()) {
+        return;
+      }
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.kind === kind);
+      const resourceObj = { ...this.state.resourceObj };
+      const template = resourceObj.spec.template;
+      let { volumeName, mountPath, subPath, claimName } = this.state.formData;
+      const readOnly = this.state.readOnly;
+      const storage = {
+        attach: null,
+        existingMountPaths: null,
+        volumeAlreadyMounted: null,
+        onPVCSelected: null,
+      };
+      storage.attach = {
+        persistentVolumeClaim: claimName,
+        volumeName: volumeName,
+        mountPath: mountPath,
+        subPath: subPath,
+        readOnly: this.state.readOnly,
+        allContainers: true,
+        containers: {},
+      };
+      const isContainerSelected = function(container) {
+        return (
+          storage.attach.allContainers ||
+          storage.attach.containers[container.name]
+        );
+      };
+
+      //   Look at the existing mount paths so that we can warn if the new value is not unique.
+      storage.existingMountPaths = getMountPaths(
+        _.get(resourceObj.spec, 'template'),
+        isContainerSelected
+      );
+      const updateVolumeName = function() {
+        const pvc = _.get(resourceObj.spec, 'volumes.persistentVolumeClaim');
+        if (!pvc) {
+          return;
+        }
+
+        // Check if there is already a volume for this PVC.
+        let volumes = _.get(resourceObj.spec, 'template.spec.volumes');
+        let volume = _.find(volumes, {
+          persistentVolumeClaim: {
+            claimName: pvc.metadata.name,
+          },
+        });
+        if (volume) {
+          // If there's already a  volume, reuse the volume name.
+          storage.attach.volumeName = resourceObj.spec.volumes.metadata.name;
+          storage.volumeAlreadyMounted = true;
+        } else if (storage.volumeAlreadyMounted) {
+          // Clear the volume name value since it was associated with the previously selected PVC.
+          storage.attach.volumeName = '';
+          storage.volumeAlreadyMounted = false;
+        }
+      };
+      storage.onPVCSelected = updateVolumeName;
+
+      // generate a volume name if not provided
+      if (!volumeName) {
+        // Create a volume name using the project
+        volumeName = `volume-${metadata.namespace}`;
```

Comment:
This seems like a weird default name and will conflict it you add multiple volumes.

Reply:
I created a random name generator.  I meant to do that before but did not get around to it.